lib/grpc_mock/grpc_stub_adapter.rb

```diff
@@ -5,19 +5,23 @@
 require 'grpc_mock/mocked_call'
 
 module GrpcMock
+  MockedOperation = Struct.new(:call, :metadata, :execute)
+
   class GrpcStubAdapter
     # To make hook point for GRPC::ClientStub
     # https://github.com/grpc/grpc/blob/bec3b5ada2c5e5d782dff0b7b5018df646b65cb0/src/ruby/lib/grpc/generic/service.rb#L150-L186
     module Adapter
-      def request_response(method, request, *args, metadata: {}, **kwargs)
+      def request_response(method, request, *args, metadata: {}, return_op: false, **kwargs)
         unless GrpcMock::GrpcStubAdapter.enabled?
           return super
         end
 
         mock = GrpcMock.stub_registry.response_for_request(method, request)
         if mock
           call = GrpcMock::MockedCall.new(metadata: metadata)
-          mock.evaluate(request, call.single_req_view)
+          response = mock.evaluate(request, call.single_req_view)
```

Comment:
when `return_op` is true, this method should be called after MockedOperation#execute is called, or raising an error mock will be triggered at unexpected timing.

Like the original implementation https://github.com/grpc/grpc/blob/fd3bd70939fb4239639fbd26143ec416366e4157/src/ruby/lib/grpc/generic/client_stub.rb#L249-L252

```rb
MockedOperation = Struct.new(:call, :metadata, :response) do
  def execute
    response.call
  end
end

...

if return_op
  response = -> { mock.evaluate(request, call.single_req_view) }
  GrpcMock::MockedOperation.new(call, metadata, response)
else
  mock.evaluate(request, call.single_req_view)
end

```

Reply:
OK, got it. I'll make similar as original implementation.